internal/cli/atlas/livemigrations/validation/create.go

```diff
@@ -0,0 +1,187 @@
+	var err error
+	opts.store, err = store.New(store.AuthenticatedPreset(config.Default()))
+	return err
+}
+
+var createTemplate = `ID	PROJECT ID	SOURCE PROJECT ID	STATUS
+{{.ID}}	{{.GroupID}}	{{.SourceGroupID}}	{{.Status}}`
+
+func (opts *CreateOpts) Run() error {
+	if err := opts.askDestinationDropConfirm(); err != nil {
+		return err
+	}
+
+	if err := opts.askPassword(); err != nil {
+		return err
+	}
+
+	createRequest := opts.newValidationCreateRequest()
+
+	r, err := opts.store.CreateValidation(opts.ConfigProjectID(), createRequest)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *CreateOpts) newValidationCreateRequest() *mongodbatlas.LiveMigration {
+	return &mongodbatlas.LiveMigration{
+		Source: &mongodbatlas.Source{
+			ClusterName:           opts.sourceClusterName,
+			GroupID:               opts.sourceProjectID,
+			Username:              opts.sourceUsername,
+			Password:              opts.sourcePassword,
+			SSL:                   &opts.sourceSSL,
+			CACertificatePath:     opts.sourceCACertificatePath,
+			ManagedAuthentication: &opts.sourceManagedAuthentication,
+		},
+		Destination: &mongodbatlas.Destination{
+			ClusterName: opts.destinationClusterName,
+			GroupID:     opts.ConfigProjectID(),
+		},
+		MigrationHosts: opts.migrationHosts,
+		DropEnabled:    &opts.destinationDropEnabled,
+	}
+}
+
+func (opts *CreateOpts) askDestinationDropConfirm() error {
+	if !opts.destinationDropEnabled {
+		return nil
+	}
+	confirmDrop := false
+	p := prompt.NewConfirm("are you sure you want to drop the destination collections?")
+
+	if err := survey.AskOne(p, &confirmDrop); err != nil {
+		return err
+	}
+
+	if !confirmDrop {
+		return errors.New("user-aborted. Not dropping destination collections")
+	}
+	return nil
+}
+
+func (opts *CreateOpts) askPassword() error {
+	if opts.sourceManagedAuthentication {
+		return nil
+	}
+	if opts.sourcePassword != "" {
+		return nil
+	}
+	p := &survey.Password{
+		Message: "Password:",
+	}
+	if err := survey.AskOne(p, &opts.sourcePassword); err != nil {
+		return err
+	}
+	if opts.sourcePassword == "" {
+		return errors.New("no password provided")
+	}
+	return nil
+}
+
+func (opts *CreateOpts) validate() error {
+	if opts.sourceManagedAuthentication && opts.sourceUsername != "" {
+		return fmt.Errorf("--%s=%t and --%s are exclusive", flag.LiveMigrationSourceManagedAuthentication, opts.sourceManagedAuthentication, flag.LiveMigrationSourceUsername)
+	}
+	if !opts.sourceManagedAuthentication && opts.sourceUsername == "" {
+		return fmt.Errorf("--%s=%t, --%s must be set", flag.LiveMigrationSourceManagedAuthentication, opts.sourceManagedAuthentication, flag.LiveMigrationSourceUsername)
+	}
+	if opts.sourceCACertificatePath != "" {
+		opts.sourceSSL = true
+	}
+	if opts.sourceSSL && opts.sourceCACertificatePath == "" {
+		return fmt.Errorf("--%s=%t, --%s must be set", flag.LiveMigrationSourceSSL, opts.sourceSSL, flag.LiveMigrationSourceCACertificatePath)
+	}
```

Comment:
[q] is this true? just double checking I'm not sure myself

Reply:
Good question! Technically no, if the cert has been signed by a trusted root authority. For on prem, these are usually self-signed certs, but it's possible that's not always true.

I'll remove the check to cover all cases, good spot.